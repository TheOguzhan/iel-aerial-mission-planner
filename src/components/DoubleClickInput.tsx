import * as React from "react";

interface DoubleClickInputProps {
  onChange: React.FormEventHandler<HTMLInputElement>;
  val: string;
  fixValues: Function;
  disabled?: boolean;
}

const DoubleClickInput: React.FunctionComponent<DoubleClickInputProps> = (
  props: DoubleClickInputProps
) => {
  const [edit, setEdit] = React.useState<boolean>(false);
  return (
    <>
      <div
        data-testid="val-displayer"
        style={{ display: !edit || props.disabled ? "block" : "none" }}
      >
        <h2 onDoubleClick={(e) => setEdit(true)} data-testid="doubleClicker">{props.val}</h2>
      </div>
      <input
        data-testid="val-input"
        style={{ display: edit && !props.disabled ? "block" : "none" }}
        className="border rounded-lg p-2  w-28 dark:border-green-600 dark:bg-green-500 border-blue-500 bg-blue-400 dark:text-white text-dark"
        value={props.val}
        onChange={props.onChange}
        onBlur={(e) => {
          setEdit(false);
          props.fixValues();
        }}
        onDoubleClick={(e) => {
          setEdit(false);
          props.fixValues();
        }}
      />
    </>
  );
};

DoubleClickInput.defaultProps = {
  disabled: false,
};

export default DoubleClickInput;
