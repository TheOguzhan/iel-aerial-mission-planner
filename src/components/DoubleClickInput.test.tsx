import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import React from "react";
import DoubleClickInput from "./DoubleClickInput";
import renderer from "react-test-renderer";

afterEach(cleanup);

test("should render double click input, but should not show input", () => {
  render(
    <DoubleClickInput
      val={"Dummy"}
      onChange={(e: React.FormEvent<HTMLInputElement>) => {
        console.log(e.currentTarget.value);
      }}
      fixValues={() => {
        console.log("Values fixed");
      }}
    />
  );

  const input = screen.getByTestId("val-input");
  const displayer = screen.getByTestId("val-displayer");

  expect(input).not.toBeVisible();
  expect(displayer).toBeVisible();
});

test("should render double click input and should show input", () => {
  render(
    <DoubleClickInput
      val={"Dummy"}
      onChange={(e: React.FormEvent<HTMLInputElement>) => {
        console.log(e.currentTarget.value);
      }}
      fixValues={() => {
        console.log("Values fixed");
      }}
    />
  );

  const input = screen.getByTestId("val-input");
  const displayer = screen.getByTestId("val-displayer");
  const doubleClicker = screen.getByTestId("doubleClicker");
  fireEvent.dblClick(doubleClicker);

  expect(displayer).not.toBeVisible();
  expect(input).toBeVisible();
});

test("should not render input", () => {
  render(
    <DoubleClickInput
      val={"Dummy"}
      onChange={(e: React.FormEvent<HTMLInputElement>) => {
        console.log(e.currentTarget.value);
      }}
      fixValues={() => {
        console.log("Values fixed");
      }}
      disabled={true}
    />
  );

  const input = screen.getByTestId("val-input");
  const displayer = screen.getByTestId("val-displayer");
  const doubleClicker = screen.getByTestId("doubleClicker");
  fireEvent.dblClick(doubleClicker);

  expect(displayer).toBeVisible();
  expect(input).not.toBeVisible();
});

test("matches snapshots", () => {
  const tree = renderer.create(
    <DoubleClickInput
      val={"Dummy"}
      onChange={(e: React.FormEvent<HTMLInputElement>) => {
        console.log(e.currentTarget.value);
      }}
      fixValues={() => {
        console.log("Values fixed");
      }}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
