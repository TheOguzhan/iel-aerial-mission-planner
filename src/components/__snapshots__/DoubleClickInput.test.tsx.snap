// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
Array [
  <div
    data-testid="val-displayer"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <h2
      data-testid="doubleClicker"
      onDoubleClick={[Function]}
    >
      Dummy
    </h2>
  </div>,
  <input
    className="border rounded-lg p-2  w-28 dark:border-green-600 dark:bg-green-500 border-blue-500 bg-blue-400 dark:text-white text-dark"
    data-testid="val-input"
    onBlur={[Function]}
    onChange={[Function]}
    onDoubleClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    value="Dummy"
  />,
]
`;
